pagerduty/resource_pagerduty_schedule.go

```diff
@@ -204,15 +210,50 @@ func resourcePagerDutyScheduleUpdate(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
-	o := &pagerduty.UpdateScheduleOptions{}
+	opts := &pagerduty.UpdateScheduleOptions{}
 
 	if v, ok := d.GetOk("overflow"); ok {
-		o.Overflow = v.(bool)
+		opts.Overflow = v.(bool)
+	}
+
+	// A schedule layer can never be removed but it can be ended.
+	// Here we determine which layer has been removed from the configuration
+	// and we mark it as ended. This is to avoid diff issues.
+	if d.HasChange("layer") {
+		oraw, nraw := d.GetChange("layer")
+
+		osl, err := expandScheduleLayers(oraw.([]interface{}))
+		if err != nil {
+			return err
+		}
+
+		nsl, err := expandScheduleLayers(nraw.([]interface{}))
+		if err != nil {
+			return err
+		}
+
+		for _, o := range osl {
+			found := false
+			for _, n := range nsl {
+				if o.ID == n.ID {
+					found = true
+				}
+			}
+
+			if !found {
+				end, err := timeToUTC(time.Now().Format(time.RFC3339))
+				if err != nil {
+					return err
+				}
+				o.End = end.String()
+				schedule.ScheduleLayers = append(schedule.ScheduleLayers, o)
+			}
```

Comment:
If I read this correctly, this logic only accounts for layers that are no longer found in the configuration. It does not seem to add new layers (layers in `nsl` but not in `osl`), nor does it account for layers in config but have had `end` added in, e.g. a user specifically ending a layer but not removing it from config